cou/apps/core.py

```diff
@@ -116,32 +144,42 @@ def _get_empty_hypervisor_step(self, unit: Unit) -> UnitUpgradeStep:
             coro=verify_empty_hypervisor(unit, self.model),
         )
 
-    def _get_enable_scheduler_step(self, unit: Unit) -> UnitUpgradeStep:
+    def _get_enable_scheduler_step(self, units: Optional[list[Unit]]) -> list[PostUpgradeStep]:
         """Get the step to enable the scheduler, so the unit can create new VMs.
 
-        :param unit: Unit to be enabled.
-        :type unit: Unit
-        :return: Step to enable the scheduler.
-        :rtype: UnitUpgradeStep
+        :param units: Units to be enabled.
+        :type units: Optional[list[Unit]]
+        :return: Steps to enable the scheduler on units
+        :rtype: list[PostUpgradeStep]
         """
-        return UnitUpgradeStep(
-            description=f"Enable nova-compute scheduler from unit: '{unit.name}'",
-            coro=self.model.run_action(
-                unit_name=unit.name, action_name="enable", raise_on_failure=True
-            ),
-        )
-
-    def _get_disable_scheduler_step(self, unit: Unit) -> UnitUpgradeStep:
-        """Get the step to disable the scheduler, so the unit cannot create new VMs.
-
-        :param unit: Unit to be disabled.
-        :type unit: Unit
-        :return: Step to disable the scheduler.
-        :rtype: UnitUpgradeStep
+        if not units:
```

Comment:
We are using same if condition in multiple places, but now I realized that it maybe an issue if I want to generate upgrade plan for app but not touching any units, so I passed `units == []`. Should we rather use `if units is None`? What do you think?

Reply:
Interesting...
I think passing `units = []` will only happen on all units of the hypervisor is upgraded and then `_check_application_target` will halt the upgrade plan generation. Howevwer, we can have a single place to check if units is None and not checking every time.

I will add this is this patch. It's a little bit out of the scope of this PR, but I don't think it's a big deal

Reply:
What do you mean by single place to check it?

Reply:
Digging a little bit it looks like that the changes required are more complex than I expected so I'll make it simple in this PR as you suggested. I've opened a issue #344 so we can better think about this

Reply:
> What do you mean by single place to check it?

As you said: `We are using same if condition in multiple places` That is what I meant by single place to check that condition.
This condition is happening because we need to check in multiple functions which upgrade strategy is used. We should think a better and more elegant solution for this later.